File: Python3/1288. Remove Covered Intervals.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x:(x[0], -x[1]))
        
        count = 0
        prev = 0
        
        for start, end in intervals:
            if end > prev:
                count += 1
                prev = end
                
        return count

File: Java/1288. Remove Covered Intervals.java
class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                return o1[0] == o2[0] ? o2[1] - o1[1]: o1[0] - o2[0];
            }
        });
        
        int count = 0, prev = 0;
        for(int[] cur: intervals) {
            if(cur[1] > prev) {
                count++;
                prev = cur[1];
            }
        }
        return count;
    }
}